parquet/src/main/java/org/apache/iceberg/parquet/MessageTypeToType.java

```diff
@@ -96,11 +96,13 @@ public Type struct(GroupType struct, List<Type> fieldTypes) {
 
   @Override
   public Type list(GroupType array, Type elementType) {
-    GroupType repeated = array.getType(0).asGroupType();
-    org.apache.parquet.schema.Type element = repeated.getType(0);
+    org.apache.parquet.schema.Type repeated = array.getType(0);
+    org.apache.parquet.schema.Type repeatedElement = array.getFields().get(0);
+    boolean isOldListElementType = ParquetSchemaUtil.isOldListElementType(repeatedElement, array.getName());
+    org.apache.parquet.schema.Type element = isOldListElementType ? repeated : repeated.asGroupType().getType(0);
```

Comment:
I think it may be cleaner to have a util method that returns the list element `Type` instead of doing all the logic here.